core/relayMsgs.go

```diff
@@ -47,7 +47,7 @@ func (r *RelayMsgs) IsMaxTx(msgLen, txSize uint64) bool {
 
 // Send sends the messages with appropriate output
 // TODO: Parallelize? Maybe?
-func (r *RelayMsgs) Send(src, dst Chain) {
+func (r *RelayMsgs) Send(src, dst Chain) ([]MsgID, []MsgID) {
```

Comment:
Sorry for very late comment, but I have reconsidered why the original `RelayMsgs::Send` didn't directly return the result and indirectly used `RelayMsgs::Succeeded`.
It is because the original author planned `RelayMsgs::Send` to be parallelized in the future, maybe. I'm not sure.
Anyway, we should obey this style at the moment.
Simply put, we should return the results (the pair of `[]MsgID`) as fields of `RelayMsgs` like `RelayMsgs::Succeeded`.

Reply:
Remove return array